coderd/database/databasefake/generator.go

```diff
@@ -0,0 +1,400 @@
+				Email:          takeFirst(orig.Email, namesgenerator.GetRandomName(1)),
+				Username:       takeFirst(orig.Username, namesgenerator.GetRandomName(1)),
+				HashedPassword: takeFirstBytes(orig.HashedPassword, []byte{}),
+				CreatedAt:      takeFirstTime(orig.CreatedAt, time.Now()),
+				UpdatedAt:      takeFirstTime(orig.UpdatedAt, time.Now()),
+				RBACRoles:      []string{},
+				LoginType:      takeFirst(orig.LoginType, database.LoginTypePassword),
+			})
+			require.NoError(t, err, "insert user")
+
+			output[name] = user
+
+		case database.Organization:
+			org, err := db.InsertOrganization(ctx, database.InsertOrganizationParams{
+				ID:          takeFirst(orig.ID, g.Lookup(name)),
+				Name:        takeFirst(orig.Name, namesgenerator.GetRandomName(1)),
+				Description: takeFirst(orig.Description, namesgenerator.GetRandomName(1)),
+				CreatedAt:   takeFirstTime(orig.CreatedAt, time.Now()),
+				UpdatedAt:   takeFirstTime(orig.UpdatedAt, time.Now()),
+			})
+			require.NoError(t, err, "insert organization")
+
+			output[name] = org
+
+		case database.Group:
+			org, err := db.InsertGroup(ctx, database.InsertGroupParams{
+				ID:             takeFirst(orig.ID, g.Lookup(name)),
+				Name:           takeFirst(orig.Name, namesgenerator.GetRandomName(1)),
+				OrganizationID: takeFirst(orig.OrganizationID, g.PrimaryOrg(ctx).ID),
+				AvatarURL:      takeFirst(orig.AvatarURL, "https://logo.example.com"),
+				QuotaAllowance: takeFirst(orig.QuotaAllowance, 0),
+			})
+			require.NoError(t, err, "insert organization")
+
+			output[name] = org
+
+		case database.ProvisionerJob:
+			job, err := db.InsertProvisionerJob(ctx, database.InsertProvisionerJobParams{
+				ID:             takeFirst(orig.ID, g.Lookup(name)),
+				CreatedAt:      takeFirstTime(orig.CreatedAt, time.Now()),
+				UpdatedAt:      takeFirstTime(orig.UpdatedAt, time.Now()),
+				OrganizationID: takeFirst(orig.OrganizationID, g.PrimaryOrg(ctx).ID),
+				InitiatorID:    takeFirst(orig.InitiatorID, uuid.New()),
+				Provisioner:    takeFirst(orig.Provisioner, database.ProvisionerTypeEcho),
+				StorageMethod:  takeFirst(orig.StorageMethod, database.ProvisionerStorageMethodFile),
+				FileID:         takeFirst(orig.FileID, uuid.New()),
+				Type:           takeFirst(orig.Type, database.ProvisionerJobTypeWorkspaceBuild),
+				Input:          takeFirstBytes(orig.Input, []byte("{}")),
+				Tags:           orig.Tags,
+			})
+			require.NoError(t, err, "insert job")
+
+			output[name] = job
+
+		case database.WorkspaceResource:
+			resource, err := db.InsertWorkspaceResource(ctx, database.InsertWorkspaceResourceParams{
+				ID:         takeFirst(orig.ID, g.Lookup(name)),
+				CreatedAt:  takeFirstTime(orig.CreatedAt, time.Now()),
+				JobID:      takeFirst(orig.JobID, uuid.New()),
+				Transition: takeFirst(orig.Transition, database.WorkspaceTransitionStart),
+				// TODO: What type to put here?
+				Type: takeFirst(orig.Type, ""),
```

Comment:
Since `WorkspaceResource.Type` is just a plain old string, I think something like `fake_resource` is probably fine.